index.js

```diff
@@ -37,14 +37,20 @@ export default class PCancelable {
 			const onResolve = value => {
 				if (this.#state !== promiseState.canceled || !onCancel.shouldReject) {
 					resolve(value);
-					this.#state = promiseState.resolved;
+
+					if (this.#state !== promiseState.canceled) {
+						this.#state = promiseState.resolved;
+					}
```

Comment:
Let's do an internal `.setState()` method that only sets the state if it's `pending`. And then use it everywhere `this.#state =` is used.